c/tskit/tables.c

```diff
@@ -10281,20 +10282,38 @@ tsk_table_collection_check_tree_integrity(const tsk_table_collection_t *self)
     }
     tsk_memset(parent, 0xff, self->nodes.num_rows * sizeof(*parent));
 
+    used_edges = tsk_malloc(num_edges * sizeof(*used_edges));
+    if (used_edges == NULL) {
```

Comment:
We can save a bit of vertical space (and coverage penalty) by putting this check in with the one for ``parent``, like `` if (parent == NULL || used_edges == NULL)``. This is a common pattern.